### ISSUE TITLE
Web - App displays upload popup along with edit photo page on double enter on photo upload in profile

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#21108]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the app
2. Open settings
3. Open profile
4. Hover on avatar and click on edit photo
5. Select any photo and press enter twice

## Expected Result:
App should display Edit photo page

## Actual Result:
App displays upload popup over edit photo page on double enter on image selection

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.29-3

**Reproducible in staging?:** y

**Reproducible in production?:** y

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/dc9c4c1c-1a9b-4ae9-b683-54a57f0bd3e0

https://github.com/Expensify/App/assets/93399543/5ae6a00c-fdb0-4a81-a9de-70347c4e0b43

**Expensify/Expensify Issue URL:**

**Issue reported by:** @dhanashree-sawant

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1686674302705589

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by alitoshmatov** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Web - App displays upload popup along with edit photo page on double enter on photo upload in profile
### What is the root cause of that problem?
When we press enter first time, file window is getting closed, and avatar is being focused. In the second time we press enter the crop modal has not been opened yet, and avatar is getting pressed showing popover menu. 
### What changes do you think we should make in order to solve the problem?
We should add ref to avatar pressable here which is getting focused in between of enters
https://github.com/Expensify/App/blob/e6c2704eb43d4d31a55eff5d5d1b3e89c5292bc3/src/components/AvatarWithImagePicker.js#L278-L282

Then we should onBlur this button when image is selected, so that it won't be pressed when second enter occurs. Here:
https://github.com/Expensify/App/blob/e6c2704eb43d4d31a55eff5d5d1b3e89c5292bc3/src/components/AvatarWithImagePicker.js#L197


### What alternative solutions did you explore? (Optional)

**Comment by muttmuure** [comment]
Unable to reproduce this. If you're still seeing it please comment in the issue

**Comment by dhanashree-sawant** [comment]
Agree, I tested it on staging and still could see the bug.

**Comment by dhanashree-sawant** [comment]
This is on current staging build

https://github.com/Expensify/App/assets/87513216/ee23a8ea-8e8d-4d09-a108-52e4c67425fe

**Comment by muttmuure** [comment]
I have reopened the issue but it will be a few days before I can test again unfortunately

**Comment by conorpendergrast** [comment]
I've reproduced on v1.3.34-1, Mac OSX Chrome

<img width="447" alt="image" src="https://github.com/Expensify/App/assets/5802477/edb61c13-3401-49be-a4b5-d96f3b4d4830">

**Comment by pradeepmdk** [comment]
can i take this task

**Comment by pradeepmdk** [comment]
Contributor details
Your Expensify account email: ppradeepd@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~0146d771ed1c776acb

**Comment by alitoshmatov** [comment]
@pradeepmdk Please read and follow [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md)

**Comment by alitoshmatov** [comment]
@fedirjh 
I think this approach does not work since `avatarCropModal` is getting triggered before the second enter registers in avatar press.
<img width="255" alt="Screenshot 2023-06-30 at 8 54 28 PM" src="https://github.com/Expensify/App/assets/59907218/f2dd1f4a-d8c9-4e77-9cfb-be17e7d84709">

https://github.com/Expensify/App/assets/59907218/b38732a2-83b2-409f-8c78-4e660d279d49

I applied it like this, did you mean it like this @Ollyws :
<img width="477" alt="Screenshot 2023-06-30 at 8 53 47 PM" src="https://github.com/Expensify/App/assets/59907218/d803e48f-bf03-4bcd-91d6-2ea18448598e">

**Comment by alitoshmatov** [comment]
@fedirjh 
I think it is taking advantage of asynchronous callback and looks like hacky since we can see popup open and close quickly.


https://github.com/Expensify/App/assets/59907218/6d216661-7830-459e-aa4a-4036d4238647

**Comment by alitoshmatov** [comment]
@fedirjh But I think popup opening and closing again is considered a bug and would be reported in future

**Comment by alitoshmatov** [comment]
I think introducing simple non-state variable to keep track of crop modal status is simpler solution, as mentioned in my alternative solutions

> What alternative solutions did you explore? (Optional)
Or we can introduce new variable like isCropModalOpen which keeps track of crop modal status. And we can prevent unwanted press event by checking its value here
https://github.com/Expensify/App/blob/e6c2704eb43d4d31a55eff5d5d1b3e89c5292bc3/src/components/AvatarWithImagePicker.js#LL279C1-L279C1
> We cannot use available state(state. isAvatarCropModalOpen) since it is not being updated imidiately.

**Comment by conorpendergrast** [comment]
I'll wait another day for anyone else to try and reproduce. If we're not able to, I suggest we close with the bug bounty. Any disagreement?

**Comment by conorpendergrast** [comment]
Ok, nice, I reproduced on Chrome on OSX (I was trying on Safari yesterday)

**Comment by conorpendergrast** [comment]
So, yes, let's fix this

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by robertjchen** [comment]
Yep, that's probably the best option at this point- let's go with @Ollyws 's proposal!

**Comment by conorpendergrast** [comment]
This looks like it should have the urgency bonus included: 

- [Assigned](https://github.com/Expensify/App/issues/21108#event-9870694402) to @Ollyws on 2023-07-20 at 8:06am GMT+1
- [Merged](https://github.com/Expensify/App/pull/23249#event-9890601789)on 2023-07-22 at 7:22am

So I'm going to add the 50% urgency bonus too

**Comment by conorpendergrast** [comment]
Payouts due:

- [x] Issue Reporter: $250 @dhanashree-sawant
- [ ] Contributor: $1000 + 50% urgency bonus = $1500 @Ollyws 
- [ ] Contributor+: $1000 + 50% urgency bonus = $1500 @fedirjh 

Eligible for 50% #urgency bonus? Yes

Upwork job is [here](https://www.upwork.com/jobs/~015dc01579b72a60e7)
